aws/credentials/ec2rolecreds/ec2_role_provider.go

```diff
@@ -111,6 +111,16 @@ func (m *EC2RoleProvider) Retrieve() (credentials.Value, error) {
 	}, nil
 }
 
+// GetIdentityDocument retrieves an identity document describing an instance
+func (m *EC2RoleProvider) GetIdentityDocument() (*EC2InstanceIdentityDocument, error) {
```

Comment:
I'm thinking the `GetIdentityDocument` might be better in the `ec2metadata` package. Since the contents of the identity doc are not really specific to the `EC2RoleProvider`. And it might be useful generically to those using the EC2Metadata client.